dependency/vault_common.go

```diff
@@ -122,6 +121,16 @@ func leaseCheckWait(s *Secret) time.Duration {
 		}
 	}
 
+	// Handle if this is an AppRole secret_id with no lease
+	if _, ok := s.Data["secret_id"]; ok && s.LeaseID == "" {
+		if expInterface, ok := s.Data["secret_id_ttl"]; ok {
+			if ttlData, err := expInterface.(json.Number).Int64(); err == nil {
+				base = int(ttlData) + 1
```

Comment:
Since this is always present, we may only want to do this if `ttlData != 0`. Otherwise the renewal will be set to `1s` on non-expiring secret ID's.

Reply:
Agreed. Updated.